### ISSUE TITLE
Can we use on our own image ? 

### ISSUE THREAD

**Issue by MilesTheProwler** [custom-humans/editable-humans#2]
Hi, I want to know about your model can use my own image as an input? Is it possible to output the obj and texture ? 

Thank you,

**Comment by custom-humans** [comment]
Hi, 

Our model focuses on editing an existing mesh (given its SMPL-X registration). But it is easy to combine with other mesh reconstruction pipelines. 

First, you may want to generate a mesh from your input photo. This can be done by other work like [ECON](https://github.com/YuliangXiu/ECON). It also gives you SMPL-X registration.

However, the generated texture is not perfect because creating a 3D mesh from a single-view image is an ill-posed problem. Therefore, you can use our pipeline to render images and edit the texture on the 2D images. 

Please let me know if you have any questions.

Thanks

**Comment by MilesTheProwler** [comment]
I have existing SMPL body obj model without texture. So can I use your model to create from my own input image for the texture and apply it on my SMPL body mesh ? If yes, how can I apply it? Could you give me some instructions? 

Thanks. 
![mesh](https://github.com/custom-humans/editable-humans/assets/134680400/3d36c10f-f9c1-4326-9bef-cdfb0d85d31f)

**Comment by custom-humans** [comment]
It sounds like our pipeline can help here, but let's first clarify your desired output:

If you're looking to apply a new texture to a SMPL mesh, we can definitely do that. Our pipeline can fit the geometry code book to the SMPL mesh. You'll just need to adjust the [input file](https://github.com/custom-humans/editable-humans/blob/main/demo.py#L36) in the line of code you're working with (And possibly tweak the evaluator a bit - I can help update a version for you if that's needed).

However, it's important to know that the output mesh won't retain the SMPL (or SMPL-X) topology, because we use a marching cube to discretize the implicit representation. Also, it won't provide a texture map as we use vertex colors.

Since your task is simple, you might not need our pipeline at all. SMPL (or SMPL-X) already has its own fixed UV texture map. In fact, there are a number of existing works that show how to colorize this texture map to create a textured SMPL body. So, that might be a simpler solution for you.

**Comment by MilesTheProwler** [comment]
Thank you for your reply. But I want to do body texture like same image input on SMPL model. I cannot find other work that can apply the texture on custom 3D body obj. If your model can do that part, could you give me some instructions ? Your repo seems to use your own training data but I am not sure about I can use it on my custom input data? 

Thanks.

**Comment by custom-humans** [comment]
Let me rephrase my question:

What is the output you would like to have?

a texture map (image) with this kind of SMPL uv?

![46127638-968fe980-c231-11e8-9ef9-f3730b9f888f](https://github.com/custom-humans/editable-humans/assets/126664174/60331b08-6ac5-4185-b6e9-ef837458a31a )

or you need a new mesh that has arbitrary topology but with the texture you want?

---
Regarding the custom data. Our method can be generalized to unseen 3D obj files.

**Comment by MilesTheProwler** [comment]
I want to have a texture like yours( colorful one). 

![image](https://github.com/custom-humans/editable-humans/assets/134680400/7f4fc790-5a03-4e40-b91d-32f28f851a65)

**Comment by custom-humans** [comment]
Okay, I guess you don't want an SMPL texture map as an output. (Correct me if I am wrong because I don't know what you mean by 'texture')

If you just want to draw something on the SMPL-X mesh without having a fixed mesh topology, that is easy. As I suggested in my previous post, you just need to change the input obj to your SMPL-X mesh [here](https://github.com/custom-humans/editable-humans/blob/main/demo.py#L36). (You might also need to tweak the evaluator a bit because it only supports textured and watertight mesh - I can help update a version for you if that's what you need).

E.g.,
```python
evaluator.fitting_3D(32, 'data/test/mesh/mesh-f00194_smpl.obj', 'data/test/mesh/mesh-f00194_smpl.obj', fit_rgb=False)
```

Then uncomment [Line 42](https://github.com/custom-humans/editable-humans/blob/main/demo.py#L42) to render a new image of your SMPL-X mesh.

Once you get the rendered image, you can draw something with Photoshop or other editing tools. Put the image in a folder following [Line 45](https://github.com/custom-humans/editable-humans/blob/main/demo.py#L45) to update the mesh. Finally, you will get your new mesh after running Line 48 and Line 51.

**Comment by MilesTheProwler** [comment]
I think you misunderstand a little bit. My model is SMPL based model and it is not SMPL-X model. I want to apply human skin texture on that model. How can I apply it?

**Comment by custom-humans** [comment]
I've done my best to provide a thorough explanation of what our pipeline can offer. 

Unfortunately, there isn't a direct, out-of-the-box solution for what you're trying to achieve. Some modifications to the pipeline will be necessary to align it with your specific needs.

**Comment by TharHtetAungg** [comment]
Hi, how did you extract the texture like this ? 
![image](https://github.com/custom-humans/editable-humans/assets/79152768/dc1acf51-317a-4c99-be4e-1967063f8061)
